test/functional/treesitter/parser_spec.lua

```diff
@@ -223,7 +225,59 @@ void ui_refresh(void)
       { 1, { { "minfunc", "identifier", 11, 12, 11, 15 }, { "min_id", "identifier", 11, 27, 11, 32 } } },
       { 4, { { "fieldarg", "identifier", 12, 17, 12, 19 } } },
       { 1, { { "minfunc", "identifier", 12, 13, 12, 16 }, { "min_id", "identifier", 12, 29, 12, 35 } } },
-      { 4, { { "fieldarg", "identifier", 13, 14, 13, 16 } } }
+      { 4, { { "fieldarg", "identifier", 13, 14, 13, 16 } } },
+    }, res)
+  end)
+
+  it('support query and iter by capture for quantifiers', function()
+    insert(test_text)
+
+    local res = exec_lua([[
+      cquery = vim.treesitter.parse_query("c", ...)
+      parser = vim.treesitter.get_parser(0, "c")
+      tree = parser:parse()[1]
+      res = {}
+      for cid, node in cquery:iter_captures(tree:root(), 0, 7, 14) do
+        -- can't transmit node over RPC. just check the name and range
+        table.insert(res, {cquery.captures[cid], node:type(), node:range()})
+      end
+      return res
+    ]], '(expression_statement (assignment_expression (call_expression)))+ @funccall')
```

Comment:
I don't see any quantifier here?
It looks like you are not testing what you want...

Reply:
`+`?

Reply:
Yes the `+` is a quantifier

Reply:
But this way you are not testing what you want to I think.
The quantifiers in tree-sitter check for adjacent nodes repeated from what I can remember.

Reply:
I'm not sure I understand what you mean. To me what you are describing is exactly what the test is showing. The query is maybe a bit complicated, so let's pretend it instead is `(expression_statement)+ @funccall`, which has the same result (I should maybe change it to that). I'm then trying to capture the three adjacent assignments
```c
width = MIN(ui->width, width);
height = MIN(ui->height, height);
foo = BAR(ui->bazaar, bazaar);
```
That would result in three `expression_statement` nodes being output, which is exactly what we see.

Also maybe it's better to look at the `iter_matches` example and not the `iter_captures` example, since the output of that didn't actually change. I just put the test there to check that it didn't break. For `iter_matches` I have checked that the test does not pass without the changes of this PR and that the expected result is in line with what tree-sitter itself i.e. the `tree-sitter` cli tool says.

Reply:
I hope this was cleared up?